pkg/e2e/infra/utils.go

```diff
@@ -422,3 +424,138 @@ func waitUntilNodesAreDeleted(client runtimeclient.Client, listOpt *runtimeclien
 		return true, nil
 	})
 }
+
+func waitUntilAllRCPodsAreReady(client runtimeclient.Client, rc *corev1.ReplicationController) error {
+	return wait.PollImmediate(e2e.RetryMedium, e2e.WaitLong, func() (bool, error) {
+		rcObj := corev1.ReplicationController{}
+		key := types.NamespacedName{
+			Namespace: rc.Namespace,
+			Name:      rc.Name,
+		}
+		if err := client.Get(context.TODO(), key, &rcObj); err != nil {
+			glog.Errorf("Error querying api RC %q object: %v, retrying...", rc.Name, err)
+			return false, nil
+		}
+		if rcObj.Status.ReadyReplicas == 0 {
+			glog.Infof("Waiting for at least one RC ready replica (%v/%v)", rcObj.Status.ReadyReplicas, rcObj.Status.Replicas)
+			return false, nil
+		}
+		glog.Infof("Waiting for RC ready replicas (%v/%v)", rcObj.Status.ReadyReplicas, rcObj.Status.Replicas)
+		if rcObj.Status.Replicas != rcObj.Status.ReadyReplicas {
+			return false, nil
+		}
+		return true, nil
+	})
+}
+
+func verifyNodeDraining(client runtimeclient.Client, targetMachine *mapiv1beta1.Machine, rc *corev1.ReplicationController) (string, error) {
+	var drainedNodeName string
+	err := wait.PollImmediate(e2e.RetryMedium, e2e.WaitLong, func() (bool, error) {
+		machine := mapiv1beta1.Machine{}
+
+		key := types.NamespacedName{
+			Namespace: targetMachine.Namespace,
+			Name:      targetMachine.Name,
+		}
+		if err := client.Get(context.TODO(), key, &machine); err != nil {
+			glog.Errorf("Error querying api machine %q object: %v, retrying...", targetMachine.Name, err)
+			return false, nil
+		}
+		if machine.Status.NodeRef == nil || machine.Status.NodeRef.Kind != "Node" {
+			glog.Error("Machine %q not linked to a node", machine.Name)
+		}
+
+		drainedNodeName = machine.Status.NodeRef.Name
+		node := corev1.Node{}
+
+		if err := client.Get(context.TODO(), types.NamespacedName{Name: drainedNodeName}, &node); err != nil {
+			glog.Errorf("Error querying api node %q object: %v, retrying...", drainedNodeName, err)
+			return false, nil
+		}
+
+		if !node.Spec.Unschedulable {
+			glog.Errorf("Node %q is expected to be marked as unschedulable, it is not", node.Name)
+			return false, nil
+		}
+
+		glog.Infof("Node %q is mark unschedulable as expected", node.Name)
+
+		pods := corev1.PodList{}
+		listOpt := &runtimeclient.ListOptions{}
+		listOpt.MatchingLabels(rc.Spec.Selector)
+		if err := client.List(context.TODO(), listOpt, &pods); err != nil {
+			glog.Errorf("Error querying api for Pods object: %v, retrying...", err)
+			return false, nil
+		}
+
+		// expecting nodeGroupSize nodes
```

Comment:
This comment talks about nodes yet we are counting pods. Is the comment out of date? Is there a 1:1 between #nodes and #pods?

Reply:
Left over, comment removed